### PR TITLE
[WIP] Adding links for glossary terms

### PR DESCRIPTION
**Pull request by EKaroune**
### Summary

To add glossary links to the text for keywords

Fixes #<NUM>

### List of changes proposed in this PR (pull-request)

- [x] add links for guide for reproducibility
- [ ] add links for guide for project design
- [ ] add links for guide for communication
- [ ] add links for guide for collaboration
 - [ ] add links for guide for ethical research

### What should a reviewer concentrate their feedback on?

- [x] check links work?
- [ ] are there any keywords that still need links?
- [x] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Review by paulowoicho — APPROVED** [review]
Looks great @EKaroune! The links all work fine!

**Comment by malvikasharan** [comment]
@EKaroune Are you happy to merge this?

**Comment by EKaroune** [comment]
I've only done a few links so I would rather finish it all first before merging. I will have some time this week to do this so I'll let you know when I have finished.

**Pull request merged**
